### ISSUE TITLE
Deep Dark Doesn't Generate

### ISSUE THREAD

**Issue by AViewFromTheTop** [Osmiooo/wildmod#148]
Broke with a recent PR.

**Comment by MerpisMe** [comment]
:(

**Comment by AViewFromTheTop** [comment]
> :(

the panorama isnt real anymore

**Comment by MerpisMe** [comment]
> @MerpisMe well that's just because the biome was commented out for some reason, removing the comment shows that the deep dark still covers the overworld whenever it generates

hmmm, seems like orcinus couldn't fix it either then :/

**Comment by AViewFromTheTop** [comment]
long-story-short, depth parameters do nothing

**Comment by MerpisMe** [comment]
> long-story-short, depth parameters do nothing

looks like we'll have to wait for fabric to actually implement the biome api then
that's gotta be what the issue is

**Comment by MerpisMe** [comment]
> > long-story-short, depth parameters do nothing
> 
> looks like we'll have to wait for fabric to actually implement the biome api then that's gotta be what the issue is

till then, we could probably get away with having sculk not replace anything but deepslate and then make the surface builder and colours identical to forests

**Comment by MerpisMe** [comment]
by the way, with the PR, have you tried going into spectator and flying down to below y0 and trying locatebiome then?

**Comment by MerpisMe** [comment]
> by the way, with the PR, have you tried going into spectator and flying down to below y0 and trying locatebiome then?

apparently it's fixed, so this is the only thing i can see being the issue

**Comment by MerpisMe** [comment]
> > by the way, with the PR, have you tried going into spectator and flying down to below y0 and trying locatebiome then?
> 
> apparently it's fixed, so this is the only thing i can see being the issue

YES, THATS DEFINITELY WHAT THE ISSUE IS @AViewFromTheTop you have to go low when using /locatebiome because the biome generates low
![image](https://user-images.githubusercontent.com/27739930/154083598-68d476cd-d5b9-4ec4-948b-403e04149ff3.png)

**Comment by AViewFromTheTop** [comment]
> > > by the way, with the PR, have you tried going into spectator and flying down to below y0 and trying locatebiome then?
> > 
> > 
> > apparently it's fixed, so this is the only thing i can see being the issue
> 
> YES, THATS DEFINITELY WHAT THE ISSUE IS @AViewFromTheTop you have to go low when using /locatebiome because the biome generates low ![image](https://user-images.githubusercontent.com/27739930/154083598-68d476cd-d5b9-4ec4-948b-403e04149ff3.png)

BRO
MOJANG WHY YOU MAKE BROKEN COMMANDS

**Comment by MerpisMe** [comment]
> > its 3d so it borke
> 
> you but it finds lush caves and dripstone caves just fine

yeah this is an issue in terralith too, it just doesn't like searching for low depth cave biomes for whatever reason

in any case, WOOOOOOOOOOOOOOO

**Comment by AViewFromTheTop** [comment]
> yeah this is an issue in terralith too, it just doesn't like searching for low depth cave biomes for whatever reason
> 
> in any case, WOOOOOOOOOOOOOOO

the panorama doesn't generate

**Comment by MerpisMe** [comment]
now we just need to figure out how to remove all water and lava features from the biome

**Comment by MerpisMe** [comment]
> > fix up the sculk patch featurer
> 
> what broke

nothing, it's just a bit messy at the moment. we need to make it a bit less patchy and have much bigger blobs on average with less gaps between them, and we also need to clean up the placement of sensors and catalysts (catalysts can place on top of sculk sensors and shriekers, and sculk sensors spawn a tiny bit too often)

**Comment by AViewFromTheTop** [comment]
> nothing, it's just a bit messy at the moment. we need to make it a bit less patchy and have much bigger blobs on average with less gaps between them, and we also need to clean up the placement of sensors and catalysts (catalysts can place on top of sculk sensors and shriekers, and sculk sensors spawn a tiny bit too often)

that's just a matter of configuring the noise parameters i added
Also I mostly fixed the problem with Catalysts above strange stuff by forcing sculk underneath if there's no block entity underneath, sadly it'll be above sensors but quite rarely

**Comment by MerpisMe** [comment]
> sadly it'll be above sensors but quite rarely

ah, oh well then

in any case, i'm extremely excited about the progress we've made on this stuff

**Comment by MerpisMe** [comment]
> ```
>     double minThresh = 0.1 * Math.cos(((average)*Math.PI)/24);
>     double maxThresh = 0.15 * Math.sin(((average)*Math.PI)/24);
>     placePatch(context, context.getOrigin(), radius, minThreshold+minThresh, maxThreshold+maxThresh);
> ```
> 
> would that work?

i dunno, how's it look?

**Comment by AViewFromTheTop** [comment]
![2022-02-15_10 00 55](https://user-images.githubusercontent.com/87103914/154089032-d6e1ad12-bc45-4a39-8411-819b76fdfbae.png)
for some reason the Deep Dark is ALWAYS extremely small, we still have to work on that
either way there's a screenshot

**Comment by MerpisMe** [comment]
> ![2022-02-15_10 00 55](https://user-images.githubusercontent.com/87103914/154089032-d6e1ad12-bc45-4a39-8411-819b76fdfbae.png) for some reason the Deep Dark is ALWAYS extremely small, we still have to work on that either way there's a screenshot

a bit closer, i think we need to increase the noise scale too

**Comment by AViewFromTheTop** [comment]
![2022-02-15_10 02 21](https://user-images.githubusercontent.com/87103914/154089282-4028de03-1da5-47e2-87b6-4931c1e92225.png)
![2022-02-15_10 02 30](https://user-images.githubusercontent.com/87103914/154089286-565fb603-d29a-4170-bddc-292d85032dff.png)

Either way with this change, some areas will have more gaps and others will be more solid

**Comment by AViewFromTheTop** [comment]
![2022-02-15_10 04 33](https://user-images.githubusercontent.com/87103914/154089769-9dd900a2-c298-4819-8b61-b1a67d4e19d6.png)
![2022-02-15_10 04 44](https://user-images.githubusercontent.com/87103914/154089772-405c5050-52ea-4777-80ed-50f77d63b595.png)
these two screenshots show it off better, actually
I'm gonna make the transition between the parameters take longer before i commit tho
